docs/scos/dev/guidelines/keeping-a-project-upgradable/upgradability-guidelines/private-api-is-used-on-the-project-level.md

```diff
@@ -128,3 +130,416 @@ class CustomerReader extends SprykerCustomerReader implements CustomerReaderInte
+    {
+        ...
+    }
+}
+```
+
+## Example of related error in the Evaluator output
+
+```bash
+------------------------------------------------------------------------------------
+Please avoid overriding of core method Spryker\Yves\CheckoutPage\CheckoutPageDependencyProvider::getCustomerStepHandler() in the class Pyz\Yves\CheckoutPage\CheckoutPageDependencyProvider
+------------------------------------------------------------------------------------
+```
+
+Make the name of the Private API entity unique by adding `Pyz`.
+
+```php
+namespace Pyz\Yves\CheckoutPage;
+
+use Spryker\Yves\CheckoutPage\CheckoutPageDependencyProvider as SprykerCheckoutPageDependencyProvider;
+
+class CheckoutPageDependencyProvider extends SprykerCheckoutPageDependencyProvider
+{
+    /**
+     * ...
+     */
+    protected function getPyzCustomerStepHandler(): ...
+    {
+        ...
+    }
+}
+```
+
+To make the names of extended Private API entities unique, you can use any other strategy. For example, you can prefix them with your project name.
+
+After renaming the entity, re-evaluate the code. The same error shouldn't be returned.
+
+
+
+## PrivateApi:Persistence
+
+It is allowed to use factory classes inside of repositories and entity manager classes on a project level, but only the methods from these classes, that were declared on the project level.
+
+### Example of code that causes an upgradability error
+
+`CustomerAccessEntityManager` uses factory method `createCustomerAccessMapper` that was declared on the vendor level.
+
+```php
+use Spryker\Zed\CustomerAccess\Persistence\CustomerAccessEntityManager as SprykerCustomerAccessEntityManager;
+
+/**
+ * @method \Pyz\Zed\Customer\Business\CustomerBusinessFactory getFactory()
+ */
+class CustomerAccessEntityManager extends SprykerCustomerAccessEntityManager implements CustomCustomerAccessEntityManager
+{
+   /**
+     * 
+     */
+    public function setContentTypesToAccessible(CustomerAccessTransfer $customerAccessTransfer): void
+    {
+        $this->getFactory()->createCustomerAccessMapper();
+       ....
+    }
+}
+```
+
+### Example of related error in the Evaluator output
+
+```bash
+------------------------------------------------------------------------------------------------------------------------
+Please avoid usage of Spryker\...\CustomerAccessEntityManager::createCustomerAccessMapper(...) in Pyz/Zed/CustomerAccess/Persistence/CustomerAccessEntityManager
+------------------------------------------------------------------------------------------------------------------------
+```
+
+### Resolving the error
+
+The solution for the error is the same as in `Resolving the error`[PrivateApi:Facade](#privateapi:facade)
+
+### Example of resolving the error by cloning functionality to the project level
+
+Copy the method to the factory on the project level.
+
+```php
+namespace Pyz\Zed\Customer\Business;
+
+use Pyz\Zed\Customer\Business\Customer\CustomerReader;
```

Comment:
Here should be `CustomerPyzReader` instead of `CustomerReader`

Reply:
++ 
Thanks